qanta/wikipedia/cached_wikipedia.py

```diff
@@ -149,7 +151,28 @@ def load_page(self, key: str):
             return CachedWikipedia.load_page(key)
         return raw
 
+    def get_wikidata(self, key: str):
+        web_page = "https://www.wikidata.org/w/api.php?action=wbsearchentities&search=" + key + "&language=en&format=json"
+        headers = {"Accept": "application/json"}
+        req = requests.get(web_page, headers=headers)
+
+        try:
+            if not req.json()['search']:
+                log.info("No results "" + search_term + " : " + search_term)
+            else:
+                id = req.json()['search'][0]['id']
+                url = req.json()['search'][0]['concepturi']
+                headers = {"Accept": "application/json"}
+                req = requests.get(url, headers=headers)
+                claims = req.json()['entities'][id]['claims']
+                return claims
```

Comment:
This is returning a dictionary that looks like this:

```
{'P1036': [{'id': 'Q22$7b6061b8-4311-099d-3d2f-3a7de56cf23a',
   'mainsnak': {'datatype': 'string',
    'datavalue': {'type': 'string', 'value': '2--411'},
    'property': 'P1036',
    'snaktype': 'value'},
   'rank': 'normal',
   'type': 'statement'}],
 'P1151': [{'id': 'Q22$04c44166-4108-c316-d451-aa9a60524349',
   'mainsnak': {'datatype': 'wikibase-item',
    'datavalue': {'type': 'wikibase-entityid',
     'value': {'entity-type': 'item',
      'id': 'Q8143311',
      'numeric-id': 8143311}},
    'property': 'P1151',
    'snaktype': 'value'},
   'rank': 'normal',
   'type': 'statement'}],
 'P1245': [{'id': 'Q22$1230f800-4cb0-aff4-6564-dd7f075411bb',
   'mainsnak': {'datatype': 'external-id',
    'datavalue': {'type': 'string', 'value': '374042'},
    'property': 'P1245',
    'snaktype': 'value'},
   'rank': 'normal',
   'type': 'statement'}],
```

Unfortunately, this isn't particularly easy to work with since the wikidata attributes appear as `P1036` rather than what they actually stand for, and the value of the attribute is buried in the dictionary, for example: `data['P1036'][0]['mainsnak']['datavalue']['value']`. Solving this will require more work in getting a list of all the attributes in wikidata and mapping them to common names. On the API a search like this filtering on the property type will do that: https://www.wikidata.org/w/api.php?action=wbsearchentities&search=doctoral%20advisor&language=en&type=property

In general though it might be better to find a way to read the wikidata dump.